Objects/regions.c

```diff
@@ -174,13 +242,21 @@ static void regionmetadata_inc_osc(Py_region_ptr_t self_ptr)
 #define regionmetadata_inc_osc(self) \
     (regionmetadata_inc_osc(REGION_PTR_CAST(self)))
 
-static void regionmetadata_dec_osc(Py_region_ptr_t self_ptr)
+static int regionmetadata_dec_osc(Py_region_ptr_t self_ptr)
 {
     if (!HAS_METADATA(self_ptr)) {
-        return;
+        return 0;
```

Comment:
Should this be called on a local or immutable region.   Could we error here?

Reply:
Yes, we could error here, but I found that every place I used this function also needed a `HAS_METADATA()` check. So having the check outside the function just spread several if statements around the codebase, while now it`s a nice `regionmetadata_dec_osc(region)` call.

So, we could error, but I don't really see an upside.